lightning/restore/restore.go

```diff
@@ -1265,7 +1265,12 @@ func (t *TableRestore) restoreEngines(ctx context.Context, rc *RestoreController
 		indexWorker := rc.indexWorkers.Apply()
 		defer rc.indexWorkers.Recycle(indexWorker)
 
-		indexEngine, err := rc.backend.OpenEngine(ctx, t.tableName, indexEngineID)
+		engineCtx := context.WithValue(ctx, kv.LocalEngineConfigKey, kv.LocalEngineConfig{
+			Compact:            true,
+			CompactConcurrency: 4,
+			CompactThreshold:   4 << 30, // 4GB
+		})
```

Comment:
it is very strange to use `context.WithValue` to pass in these settings. since we control the interface of `OpenEngine` we could add an argument specifying the type of engine (index engine, data engine sorted, data engine unsorted).